tokio-trace/tokio-trace-core/src/callsite.rs

```diff
@@ -24,23 +24,44 @@ struct Registry {
     dispatchers: Vec<dispatcher::Registrar>,
 }
 
+impl Registry {
+    fn rebuild_callsite_interest(&self, callsite: &'static Callsite) {
+        let meta = callsite.metadata();
+
+        let mut interest = Interest::never();
+
+        for registrar in &self.dispatchers {
+            if let Some(sub_interest) = registrar.try_register(meta) {
+                interest = interest.and(sub_interest);
+            }
+        }
+
+        callsite.set_interest(interest)
+    }
+
+    fn rebuild_interest(&mut self) {
+        self.dispatchers.retain(Registrar::is_alive);
+
+        self.callsites.iter().for_each(|&callsite| {
+            self.rebuild_callsite_interest(callsite);
+        });
+    }
+}
+
 /// Trait implemented by callsites.
+///
+/// These functions are only intended to be called by the [`Registry`] which
+/// correctly handles determining the common interest between all subscribers.
 pub trait Callsite: Sync {
-    /// Adds the [`Interest`] returned by [registering] the callsite with a
-    /// [dispatcher].
-    ///
-    /// If the interest is greater than or equal to the callsite's current
-    /// interest, this should change whether or not the callsite is enabled.
+    /// Sets the [`Interest`] for this callsite.
     ///
     /// [`Interest`]: ../subscriber/struct.Interest.html
-    /// [registering]: ../subscriber/trait.Subscriber.html#method.register_callsite
-    /// [dispatcher]: ../dispatcher/struct.Dispatch.html
-    fn add_interest(&self, interest: Interest);
+    fn set_interest(&self, interest: Interest);
 
-    /// Remove _all_ [`Interest`] from the callsite, disabling it.
+    /// Gets the [`Interest`] for this callsite.
     ///
     /// [`Interest`]: ../subscriber/struct.Interest.html
-    fn clear_interest(&self);
+    fn get_interest(&self) -> Interest;
```

Comment:
Nit: I think the Rust idiom is that accessor methods generally don't include the word "get": https://rust-lang-nursery.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter

```suggestion
    fn interest(&self) -> Interest;
```

Reply:
Also, I know that in the past, @carllerche [was opposed to having a method like this](https://github.com/hawkw/tokio-trace-prototype/issues/120) on the callsite trait. I see why it's necessary now (to allow `register` to be implemented using `set_interest` rather than `add_interest`). 

Carl, do you think it's better to have `interest` and `set_interest`, or separate `set_interest` and `add_interest` functions in this API?